### PR TITLE
drivers: udc_dwc2: Fix timeouts when disabling endpoints

### PR DESCRIPTION
**Pull request by tmon-nordic**
Do not set NAK bit again for endpoints that already have NAK bit set. Do not wait for OUT endpoint 0 disable because it cannot be disabled by application (DOEPCTL0 EPDis bit is Read-Only).

Disable endpoints before disabling interrupts because it is necessary to handle RXFLVL interrupt (in Slave mode) for GOUTNAKEFF to become active.

### CODE REVIEW

File: drivers/usb/udc/udc_dwc2.c [hunk 1]
```diff
@@ -1725,6 +1737,9 @@ static int udc_dwc2_disable(const struct device *dev)
 		return -EIO;
 	}
 
+	config->irq_disable_func(dev);
+	sys_clear_bits((mem_addr_t)&base->gahbcfg, USB_DWC2_GAHBCFG_GLBINTRMASK);
+
```
Comment (jfischer-no) [inline drivers/usb/udc/udc_dwc2.c:1735]: The other changes in udc_dwc2_ep_disable() look like f4e95ccd48e51c6f8ea351de1fd2e32955d4ce99 fixes. Perhaps this should be moved to its own commit.

File: drivers/usb/udc/udc_dwc2.c [hunk 2]
```diff
@@ -1258,6 +1258,15 @@ static void udc_dwc2_ep_disable(const struct device *dev,
 	dxepctl_reg = dwc2_get_dxepctl_reg(dev, cfg->addr);
 	dxepctl = sys_read32(dxepctl_reg);
 
+	if (dxepctl & USB_DWC2_DEPCTL_NAKSTS) {
+		/* Endpoint already sends forced NAKs. STALL if necessary. */
+		if (stall) {
+			dxepctl |= USB_DWC2_DEPCTL_STALL;
+			sys_write32(dxepctl, dxepctl_reg);
+		}
+		return;
```
Comment (jfischer-no) [inline drivers/usb/udc/udc_dwc2.c:1222]:
```c
		}

		return;
```

File: drivers/usb/udc/udc_dwc2.c (hunk 1 above)
Comment (tmon-nordic) [reply drivers/usb/udc/udc_dwc2.c:1735]: Yes, the changes here are really fixes to f4e95ccd48e51c6f8ea351de1fd2e32955d4ce99. I don't think this commit should be splitted.

**Review by jfischer-no — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
